### PR TITLE
feat: Add LSP7Votes and LSP8Votes extension to support LSP-based Governance

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?
## 🚀 Feature

- Add implementation of LSP7 and LSP8 required interface for supporting governance compliant to ERC5805.

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: packages/lsp7-contracts/contracts/LSP7DigitalAssetCore.sol [hunk 1]
```diff
@@ -441,19 +441,7 @@ abstract contract LSP7DigitalAssetCore is ILSP7DigitalAsset {
...
-
-        emit Transfer({
-            operator: msg.sender,
-            from: address(0),
-            to: to,
-            amount: amount,
-            force: force,
-            data: data
-        });
+        _update(address(0), to, amount, force, data);
```
Comment (CJ42) [inline packages/lsp7-contracts/contracts/LSP7DigitalAssetCore.sol:444]: Let's wait for #974 to be merged first, as the `Core` contract does not exist anymore,

File: packages/lsp7-contracts/contracts/LSP7DigitalAssetCore.sol [hunk 2]
```diff
@@ -614,29 +586,64 @@ abstract contract LSP7DigitalAssetCore is ILSP7DigitalAsset {
...
+    ) internal virtual {
+        if (from == address(0)) {
+            // Overflow check required: The rest of the code assumes that totalSupply never overflows
+            _existingTokens += value;
+        } else {
+            uint256 fromBalance = _tokenOwnerBalances[from];
+            if (fromBalance < value) {
+                revert LSP7AmountExceedsBalance(fromBalance, from, value);
+            }
+            unchecked {
```
Comment (CJ42) [inline packages/lsp7-contracts/contracts/LSP7DigitalAssetCore.sol:621]: Nice this is inspired from OZ 🫡

File: packages/lsp7-contracts/contracts/extensions/LSP7Votes.sol [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return _domainSeparatorV4();
+    }
+
+    /**
+     * @dev Get the `pos`-th checkpoint for `account`.
+     */
+    function checkpoints(
+        address account,
+        uint32 pos
```
Comment (CJ42) [inline packages/lsp7-contracts/contracts/extensions/LSP7Votes.sol:67]: What does `pos` stands for? Suggest a more explicit param name

**Review by CJ42 — APPROVED** [review]

**Review by skimaharvey — APPROVED** [review]
LGTM

**Review by CJ42 — COMMENTED** [review]
We should create the `InitAbstract` version as well.

**Review by CJ42 — APPROVED** [review]

**Pull request merged**